transactions/stakingCollection/register_node.cdc

```diff
@@ -8,22 +8,27 @@ transaction(id: String,
             networkingAddress: String,
             networkingKey: String,
             stakingKey: String,
-            amount: UFix64) {
+            amount: UFix64,
+            publicKeys: [[UInt8]]) {
     
     let stakingCollectionRef: &FlowStakingCollection.StakingCollection
 
     prepare(account: AuthAccount) {
         self.stakingCollectionRef = account.borrow<&FlowStakingCollection.StakingCollection>(from: FlowStakingCollection.StakingCollectionStoragePath)
             ?? panic("Could not borrow ref to StakingCollection")
-    }
 
-    execute {
-        self.stakingCollectionRef.registerNode(
+        if let machineAccount = self.stakingCollectionRef.registerNode(
             id: id,
             role: role,
             networkingAddress: networkingAddress,
             networkingKey: networkingKey,
             stakingKey: stakingKey,
-            amount: amount)
+            amount: amount,
+            payer: account) 
+        {
+            for key in publicKeys {
+                machineAccount.addPublicKey(key)
+            }
```

Comment:
May also want to add a sanity check that there is at least one key here - otherwise the machine account will be unusable.

Reply:
Good idea. I'll add that